tools/plugins/com.liferay.ide.upgrade.plan.ui/src/com/liferay/ide/upgrade/plan/ui/internal/UpgradePlanLabelProvider.java

```diff
@@ -49,6 +57,30 @@ public Image getImage(Object element) {
 			return UpgradePlanUIPlugin.getImage(UpgradePlanUIPlugin.NO_TASKS_IMAGE);
 		}
 
+		Image baseImage = super.getImage(element);
+
+		UpgradeTaskStepAction stepAction = Adapters.adapt(element, UpgradeTaskStepAction.class);
+
+		if (stepAction != null) {
+			ImageDescriptor statusImageDescriptor = null;
+
+			Bundle bundle = FrameworkUtil.getBundle(element.getClass());
+
+			if (UpgradeTaskStepActionStatus.COMPLETED.equals(stepAction.getStatus())) {
+				statusImageDescriptor = ImageDescriptor.createFromURL(
```

Comment:
The urls to the complete_status and skip_status should be known and pulls from a constant pool instead of dynamically found and built each time.